src/rpdk/guard_rail/rule_library/core/schema-linter-core-rules.guard

```diff
@@ -27,67 +20,116 @@ rule ensure_primary_identifier_exists_and_not_empty
+                }
+                >>
+            }
+        }
+    }
+
+    when readOnlyProperties !exists {
+        createOnlyProperties !exists
         <<
         {
             "result": "NON_COMPLIANT",
             "check_id": "PID003",
             "message": "primaryIdentifier MUST be either readOnly or createOnly"
         }
         >>
+
+        when createOnlyProperties exists {
+            let create = createOnlyProperties
+            primaryIdentifier[*] {
+                this IN %create
+                <<
+                {
+                    "result": "NON_COMPLIANT",
+                    "check_id": "PID003",
+                    "message": "primaryIdentifier MUST be either readOnly or createOnly"
+                }
+                >>
+            }
+        }
     }
 }
 
 
 
 rule ensure_create_and_read_only_intersection_is_empty
 {
-    readOnlyProperties[*] {
-        this !IN %create
-        <<
-        {
-            "result": "NON_COMPLIANT",
-            "check_id": "PR001",
-            "message": "read/createOnlyProperties MUST NOT have common properties"
+    when readOnlyProperties exists {
+        when createOnlyProperties exists {
+            let create = createOnlyProperties
+            readOnlyProperties[*] {
+                this !IN %create
+                <<
+                {
+                    "result": "NON_COMPLIANT",
+                    "check_id": "PR001",
+                    "message": "read/createOnlyProperties MUST NOT have common properties"
+                }
+                >>
+            }
         }
-        >>
-    }
-
-    createOnlyProperties[*] {
-        this !IN %read
-        <<
-        {
-            "result": "NON_COMPLIANT",
-            "check_id": "PR002",
-            "message": "create/readOnlyProperties MUST NOT have common properties"
-        }
-        >>
     }
 }
 
 rule ensure_write_and_read_only_intersection_is_empty
 {
-    readOnlyProperties[*] {
-        this !IN %write
-        <<
-        {
-            "result": "NON_COMPLIANT",
-            "check_id": "PR003",
-            "message": "read/writeOnlyProperties MUST NOT have common properties"
+    when readOnlyProperties exists {
+        when writeOnlyProperties exists {
+            let write = writeOnlyProperties
+            readOnlyProperties[*] {
+                this !IN %write
+                <<
+                {
+                    "result": "NON_COMPLIANT",
+                    "check_id": "PR004",
+                    "message": "write/readOnlyProperties MUST NOT have common properties"
+                }
+                >>
+            }
         }
-        >>
     }
-    writeOnlyProperties[*] {
-        this !IN %read
-        <<
-        {
-            "result": "NON_COMPLIANT",
-            "check_id": "PR004",
-            "message": "write/readOnlyProperties MUST NOT have common properties"
+
+    when writeOnlyProperties exists {
```

Comment:
Same comment here, this condition is the same as the last but reversed. I don't think it is necessary